feat: add logger with Alma SBOM format (#6)

redadmin-k opened the pull request:
This PR implements a structured logger for the alma-sbom project as requested in issue #6.

- Logger initialized in `alma_sbom/__init__.py`
- Log format includes "Alma SBOM"
- Example usage added to `type.py`

KAWAHARA-souta commented:
Hi, Thank you for sending PR.
I am not a maintainer, but I'd like to leave a few comments.

Actually, Logger initialization is already implemented, although not completely, in https://github.com/AlmaLinux/alma-sbom/blob/main/alma_sbom/cli/logging.py.
I think your updates to alma_sbom/init.py should be included there.

Currently, we get logger instances in each module as follows:
https://github.com/AlmaLinux/alma-sbom/blob/main/alma_sbom/cli/main.py#L11
```
_logger = getLogger(__name__)
```

Outputting module loading logs is good, but I think they should at least be debug outputs.

redadmin-k commented:
Thanks for your review, @KAWAHARA-souta.

I've updated the PR:
- Removed logger initialization from `__init__.py`
- Centralized logger setup in `cli/logging.py` via `setup_logger()`
- Updated `main.py` to use consistent logger pattern

Let me know if any further changes are needed!

KAWAHARA-souta commented:
Hmm, I'm sorry. It seems I haven't been able to convey my intention well.


The Logger initialization has already been implemented in `Logging.setup()`.  (Please see L.14~L.19)
So there is no need to implement `setup_logger()`.

In the current implementation, the root logger is set up simply by using logging.basicConfig(). This is because any logger instances created thereafter will always inherit from the root logger, eliminating the need for individual configuration.

Your implementation configures each logger instance individually every time it's generated in each module, which cannot be considered an optimal implementation.

If you've found any issues that need to be fixed in the existing Logging.setup() implementation, please fix those instead.

redadmin-k commented:
Thanks for the clarification, @KAWAHARA-souta.

I've removed the `setup_logger()` function and reverted the logger usage in `main.py` back to `getLogger(__name__)`, as the `Logging.setup()` already handles root logger configuration.

If there's anything else you'd like me to address, feel free to let me know!

javihernandez commented:
Hey @redadmin-k, welcome thanks for your willing to contribute to alma-sbom! As @KAWAHARA-souta already pointed out (thanks Souta!), this issue was initially resolved as part of https://github.com/AlmaLinux/alma-sbom/pull/31, and in my opinion I don't think there's anything else to do here. We should just close the issue and try finding something else for you to work on, what do you think?
